net-vpn/headscale/headscale-0.15.0.ebuild

```diff
@@ -0,0 +1,47 @@
+# Copyright 2020-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+inherit go-module systemd
+
+GENTOO_DEPEND_ON_PERL=no
+
+DESCRIPTION="An open source, self-hosted implementation of the Tailscale control server."
+HOMEPAGE="https://github.com/juanfont/headscale"
+SRC_URI="
+	https://github.com/juanfont/headscale/archive/refs/tags/v0.15.0.tar.gz -> ${P}.tar.gz
+	https://dev.gentoo.org/~jsmolic/distfiles/${P}-deps.tar.xz
+"
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64"
+
+DEPEND="
+	acct-user/headscale
+	acct-group/headscale
+"
+
+RDEPEND="net-firewall/iptables"
+
+src_compile() {
+	if [[ -f Makefile ]] || [[ -f GNUmakefile ]] || [[ -f makefile ]]; then
```

Comment:
This is the [default src_compile implementation](https://devmanual.gentoo.org/ebuild-writing/functions/src_compile/index.html) which means it will be executed by default, so there is no need to repeat it here

Reply:
To explain a bit more about this, this is the default implementation because Makefile-based build systems are the most common, so we cover many packages with this impl, but when you start packaging some software, you already know what build system are you dealing with (because you verified it by looking at the source code ;-)), e.g. cmake, meson, autotools, etc. and you pick the build eclass accordingly (meson.eclass, cmake.eclass) which have their own implementations of common phase functions